projects/clr/rocclr/device/rocm/rocdevice.cpp

```diff
@@ -2058,6 +2059,23 @@ void* Device::hostNumaAlloc(size_t size, size_t alignment, MemorySegment mem_seg
       ptr = hostAlloc(size, alignment, mem_seg, cpu_agent_info_);
   }
   numa_free_cpumask(nodeMask);
+#else // _WIN32
+  // This depends on affinity set by the user or Os default
+  PROCESSOR_NUMBER procNumber{};
```

Comment:
- We should not query this info for every host call. You should abstract it.
- Also try to move the abstraction under OS module.
- Could you get rid of numa lib and implement similar logic in both paths.  I hardly understand this logic and why Windows path doesn't need it.
  switch (mode) {
    // For details, see "man get_mempolicy".
    case MPOL_BIND:
    case MPOL_PREFERRED:
      // We only care about the first CPU node